# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/push/PushManager.cpp
# Commit: ac548c2a725b
# Full Hash: ac548c2a725b66db57c81bb24accca2bbe075931
# Author: Olli Pettay <Olli.Pettay@helsinki.fi>
# Date: 2019-07-29 21:38:39
# Description:
#   Bug 1565930, try to make Promise less error prone to compartment mismatches, r=bzbarsky
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D38762
# ==============================================================================

diff -r c4980c083852 -r ac548c2a725b dom/push/PushManager.cpp
--- a/dom/push/PushManager.cpp	Mon Jul 29 08:00:35 2019 +0000
+++ b/dom/push/PushManager.cpp	Mon Jul 29 14:43:54 2019 +0000
@@ -299,7 +299,7 @@
     if (NS_SUCCEEDED(mStatus)) {
       promise->MaybeResolve(mState);
     } else {
-      promise->MaybeReject(aCx, JS::UndefinedHandleValue);
+      promise->MaybeRejectWithUndefined();
     }
 
     mProxy->CleanUp();
@@ -437,7 +437,7 @@
 
   RefPtr<PromiseWorkerProxy> proxy = PromiseWorkerProxy::Create(worker, p);
   if (!proxy) {
-    p->MaybeReject(worker->GetJSContext(), JS::UndefinedHandleValue);
+    p->MaybeRejectWithUndefined();
     return p.forget();
   }
 